types/timsort/index.d.ts

```diff
@@ -0,0 +1,33 @@
+// Type definitions for timsort 0.3
+// Project: https://github.com/mziccard/node-timsort
+// Definitions by: Vunovati <https://github.com/Vunovati>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ *
+ * @param array - The array to sort.
+ * @param compare - Item comparison function. Default is alphabetical
+ */
+export function sort<T>(array: T[], compare?: (a: T, b: T) => number): void;
+
+/**
+ *
+ * @param array - The array to sort.
+ * @param lo - First element in the range (inclusive). Default is 0
+ * @param hi - Last element in the range. Default is last element
+ */
+export function sort(array: any[], lo?: number, hi?: number): void;
```

Comment:
Is it possible to have just 2 overloads (in this order):
export function sort(array: any[], lo: number, hi?: number): void; // If 2nd arg is number then we know no comparer was passed in
export function sort<T>(array: T[], compare?: (a: T, b: T) => number, lo?: number, hi?: number): void;